src/components/Chat.js

```diff
@@ -0,0 +1,214 @@
+    placeholderStyle={styles.placeholderStyle}
+    rounded
+    source={uri ? { uri } : undefined}
+    renderPlaceholderContent={
+      <Avatar
+        containerStyle={[styles.containerStyle]}
+        overlayContainerStyle={[styles.overlayContainerStyle, styles.border]}
+        rounded
+        title={title}
+        titleStyle={styles.titleStyle}
+      />
+    }
+  />
+);
+
+/**
+ * it is the component used to realise the chat function
+ * @param {array} data please make sure that the data format is as shown in the document
+ *                      https://github.com/FaridSafi/react-native-gifted-chat#message-object
+ * @param {object} bubbleWrapperStyleLeft  style of chat balloons on the left
+ * @param {object} bubbleWrapperStyleRight style of chat balloons on the right
+ * @param {object} messageTextStyleLeft    style of chat text on the left
+ * @param {object} messageTextStyleRight   style of chat text on the right
+ * @param {func}   onSendButton            function returning message text
+ * @param {string} placeholder             placeholder text of `textInput`
+ * @param {object} textInputProps          props to customise text input
+ * @param {number} userId                  prop to recognise whether the message is the owner
+ *                                         or another user
+ */
+export const Chat = ({
+  bubbleWrapperStyleLeft,
+  bubbleWrapperStyleRight,
+  data,
+  messageTextStyleLeft,
+  messageTextStyleRight,
+  onSendButton,
+  placeholder = 'Type a message',
+  textInputProps,
+  userId = 1
+}) => {
+  const [messages, setMessages] = useState(data);
+
+  useEffect(() => {
+    setMessages(data);
+  }, [data]);
+
+  const onSend = useCallback((messages) => {
+    onSendButton(messages[0].text);
+  }, []);
+
+  return (
+    <GiftedChat
+      alwaysShowSend
+      messages={messages}
+      minInputToolbarHeight={normalize(96)}
+      onSend={(messages) => onSend(messages)}
+      placeholder={placeholder}
+      scrollToBottom
+      user={{ _id: parseInt(userId) }}
+      renderAvatar={(props) => (
+        <UserAvatar
+          uri={props?.currentMessage?.user?.avatar}
+          title={props?.currentMessage?.user?.name}
+        />
+      )}
+      renderBubble={(props) => (
+        <Bubble
+          {...props}
+          wrapperStyle={{
+            left: bubbleWrapperStyleLeft || {
+              backgroundColor: colors.gray20
+            },
+            right: bubbleWrapperStyleRight || {
+              // TODO: added manually because there is no similar colour in the colours file.
+              //       can be edited later!
+              backgroundColor: '#E8F1E9'
+            }
+          }}
+        />
+      )}
+      renderComposer={(props) => (
+        <Composer
+          {...props}
+          textInputStyle={styles.textInputStyle}
+          textInputProps={textInputProps}
+        />
+      )}
+      renderDay={() => null}
```

Comment:
wow i just saw this and took it.
so we can shorten the messages to only render time, because date is there as kind of title

<img width="371" alt="image" src="https://user-images.githubusercontent.com/1942953/190173176-463eebf6-2fa0-440f-8c9e-0aeb56da1acc.png">